src/browser-frontend/views/browser/window.jsx

```diff
@@ -10,17 +10,48 @@ CONDITIONS OF ANY KIND, either express or implied. See the License for the
 specific language governing permissions and limitations under the License.
 */
 
-import React, { PureComponent } from 'react';
+import React, { PureComponent, PropTypes } from 'react';
 import CSSModules from 'react-css-modules';
+import { connect } from 'react-redux';
+import Mousetrap from 'mousetrap';
+
+import { client } from '../../global';
+import PagesModelActions from '../../actions/pages-model-actions';
+import * as SharedPropTypes from '../../model/shared-prop-types';
+import SharedActions from '../../../shared/actions/shared-actions';
+import * as UIPagesSelectors from '../../selectors/ui-pages-selectors';
+
 
 import Styles from './window.css';
 import Chrome from './chrome';
 import Content from './content';
 
+@connect(state => ({
+  client,
+  selectedPageId: UIPagesSelectors.getSelectedPageId(state),
+}))
 @CSSModules(Styles, {
   allowMultiple: true,
 })
 export default class Window extends PureComponent {
+  componentDidMount() {
+    this.mousetrap = Mousetrap();
+    this.mousetrap.bind('mod+t', () => {
+      this.props.dispatch(PagesModelActions.addPage());
+    });
+    this.mousetrap.bind('mod+q', () => {
+      this.props.client.send(SharedActions.events.fromFrontend.toServer.app.window.requestedClose());
+    });
+    this.mousetrap.bind('mod+w', () => {
+      const pageId = this.props.selectedPageId;
+      this.props.dispatch(PagesModelActions.removePage({ pageId }));
```

Comment:
As a followup, we should move all of this stuff into sagas to make them testable.